packages/amplify-graphql-transformer-core/src/transformation/transformer-plugin-base.ts

```diff
@@ -51,6 +54,19 @@ export abstract class TransformerPluginBase implements TransformerPluginProvider
     // Transformers can define extra shapes that can be used by the directive
     // and validated. TODO: Validation.
     this.typeDefinitions = extraDefinitions;
+    this.logs = [];
+  }
+
+  protected log(level: TransformerLogLevel, message: string) {
+    this.logs.push({ level, message });
+  }
```

Comment:
Perhaps vend a 'warn', 'info', etc. to reflect the log levels? Instead of passing it in at each step.